Automattic/jetpack
# VideoPress: handle poster image generation state

This PR polls the poster image generation state changing the UI appearance according to it. For this:

* it adds two new poster lib helpers: 
  * hasVideoPosterGenerated()
  * pollGeneratingPosterImage
* Request the generating state after the process is triggered
* Store the generating status in a local state of the useSyncData() hook
* Change the UI according to this state

Fixes https://github.com/Automattic/jetpack/issues/29760

## Proposed changes:

* VideoPress: handle poster image generation state

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Go to the block editor
* Create/edit a VideoPress video block instance
* Set a poster by picking a video frame
* Confirm the app updates the UI of the Poster panel (disables the control, shows a spinner, etc).
<img width="291" alt="image" src="https://user-images.githubusercontent.com/77539/228266607-9f2b648a-5f0a-43d2-9251-8c295d792d63.png">
* Confirm the app starts to poll the generating state:
<img width="852" alt="image" src="https://user-images.githubusercontent.com/77539/228266965-3b6bffea-b9a6-43cf-bbdd-9c9a0a27ea60.png">
* Confirm the app updates again the once the poster has been generated

We'll handle refreshing the player in a follow-up one.
